includes/Documents/OrderDocument.php

```diff
@@ -1367,11 +1367,91 @@ public function shop_coc_number() {
 		echo esc_html( $this->get_shop_coc_number() );
 	}
 
+	/**
+	 * Return/Show shop/company address line 1 if provided.
+	 */
+	function get_shop_address_line_1(): string {
+		return $this->get_settings_text( 'shop_address_line_1' );
+	}
+	function shop_address_line_1(): void {
+		echo esc_html( $this->get_shop_address_line_1() );
+	}
+
+	/**
+	 * Return/Show shop/company address line 2 if provided.
+	 */
+	function get_shop_address_line_2(): string {
+		return $this->get_settings_text( 'shop_address_line_2' );
+	}
+	function shop_address_line_2(): void {
+		echo esc_html( $this->get_shop_address_line_2() );
+	}
+
+	/**
+	 * Return/Show shop/company address country if provided.
+	 */
+	function get_shop_address_country(): string {
+		return $this->get_settings_text( 'shop_address_country' );
+	}
+	function shop_address_country(): void {
+		echo esc_html( $this->get_shop_address_country() );
+	}
+
+	/**
+	 * Return/Show shop/company address state if provided.
+	 */
+	function get_shop_address_state(): string {
+		return $this->get_settings_text( 'shop_address_state' );
+	}
+	function shop_address_state(): void {
+		echo esc_html( $this->get_shop_address_state() );
+	}
+
+	/**
+	 * Return/Show shop/company address city if provided.
+	 */
+	function get_shop_address_city(): string {
+		return $this->get_settings_text( 'shop_address_city' );
+	}
+	function shop_address_city(): void {
+		echo esc_html( $this->get_shop_address_city() );
+	}
+
+	/**
+	 * Return/Show shop/company address postcode if provided.
+	 */
+	function get_shop_address_postcode(): string {
+		return $this->get_settings_text( 'shop_address_postcode' );
+	}
+	function shop_address_postcode(): void {
+		echo esc_html( $this->get_shop_address_postcode() );
+	}
+
+	/**
+	 * Return/Show shop/company address additional info if provided.
+	 */
+	function get_shop_address_additional(): string {
+		return $this->get_settings_text( 'shop_address_additional' );
+	}
+	function shop_address_additional(): void {
+		echo esc_html( $this->get_shop_address_additional() );
+	}
+
 	/**
 	 * Return/Show shop/company address if provided
 	 */
-	public function get_shop_address() {
-		return $this->get_settings_text( 'shop_address' );
+	public function get_shop_address(): string {
+		$address = array_filter( array(
+			$this->get_settings_text( 'shop_address_line_1', '', false),
+			$this->get_settings_text( 'shop_address_line_2', '', false),
+			$this->get_settings_text( 'shop_address_city', '', false),
+			$this->get_settings_text( 'shop_address_postcode', '', false),
+			$this->get_settings_text( 'shop_address_state', '', false),
+			$this->get_settings_text( 'shop_address_country', '', false),
+			$this->get_settings_text( 'shop_address_additional', '', false),
+		) );
+
+		return wpautop( implode( '<br>', $address ) );
```

Comment:
I'm not sure about this. For example, in Portugal, the postcode and city are usually on the same line. Maybe you should use the `get_address_formats()` method from WooCommerce Countries?